filebeat/docs/running-on-kubernetes.asciidoc

```diff
@@ -135,3 +135,115 @@ filebeat   32        32        0         32           0           <none>
 
 Log events should start flowing to Elasticsearch. The events are annotated with
 metadata added by the <<add-kubernetes-metadata>> processor.
+
+
+[float]
+==== Parsing json logs
+
+It is common case when collecting logs from workloads running on Kubernetes that these
+applications are logging in json format. In these case, special handling can be applied so as to
+parse these json logs properly and decode them into fields. Bellow there are provided 3 different ways
```

Comment:
I think we should be more opinionated here, adding just 2 options. One for template based autodiscover and another one for hints based autodiscover. I would go for `json.*` as it gives the user more freedom down the line (multiline/line filtering happens after JSON decoding).